fabric-resource-loader-v0/src/client/java/net/fabricmc/fabric/mixin/resource/loader/client/ClientDataPackManagerMixin.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.resource.loader.client;
+
+import java.util.List;
+
+import com.llamalad7.mixinextras.injector.ModifyReturnValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.client.resource.ClientDataPackManager;
+import net.minecraft.registry.VersionedIdentifier;
+import net.minecraft.resource.ResourcePackManager;
+
+import net.fabricmc.fabric.impl.resource.loader.ModResourcePackCreator;
+import net.fabricmc.fabric.impl.resource.loader.ModResourcePackUtil;
+
+@Mixin(ClientDataPackManager.class)
+public class ClientDataPackManagerMixin {
+	@Unique
+	private static final Logger LOGGER = LoggerFactory.getLogger("ClientDataPackManagerMixin");
+
+	@Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Lnet/minecraft/resource/VanillaDataPackProvider;createClientManager()Lnet/minecraft/resource/ResourcePackManager;"))
+	public ResourcePackManager createClientManager() {
+		return ModResourcePackUtil.createClientManager();
+	}
+
+	@ModifyReturnValue(method = "getCommonKnownPacks", at = @At("RETURN"))
+	List<VersionedIdentifier> getCommonKnownPacksReturn(List<VersionedIdentifier> original) {
+		if (original.size() > ModResourcePackCreator.MAX_KNOWN_PACKS) {
+			LOGGER.warn("Too many knownPacks: Found {}; max {}. Increase -Dfabric-resource-loader-v0:maxKnownPacks to prevent unnecesary registry syncing.", original.size(), ModResourcePackCreator.MAX_KNOWN_PACKS);
```

Comment:
```suggestion
			LOGGER.warn("Too many knownPacks: Found {}; max {}", original.size(), ModResourcePackCreator.MAX_KNOWN_PACKS);
```

I dont think suggesting users to set a system prop is the solution here, as it would need changing on both sides, and cannot be done in mod packs. If this does become an issue we might need to raise it, i imagine its unlikely?

Reply:
I think the warning is more for the case of "someone decreased it, but decreased it too much", so then changing it back on that side is all you need. I don't think a scenario with more than 1024 packs is at all likely.

But I'm happy to remove the suggestion.

Reply:
I think the sysytem prop should be for developers not users. As I said we can keep an eye on this to see if people do hit the limit or have issues and possibly think again.